tensorflow/core/api_def/base_api/api_def_GenerateBoundingBoxProposals.pbtxt

```diff
@@ -0,0 +1,86 @@
+op {
+  graph_op_name: "GenerateBoundingBoxProposals"
+  in_arg {
+    name: "scores"
+    description: <<END
+A 4-D float tensor of shape `[num_images, height, width, num_achors]` containing scores of the boxes for given anchors.
+END
+  }
+  in_arg {
+    name: "bbox_deltas"
+    description: <<END
+A 4-D float tensor of shape `[num_images, height, width, 4 x num_anchors]`. encoding boxes with respec to each anchor.
+Coordinates are given in the form [dy, dx, dh, dw].
+END
+  }
+  in_arg {
+    name: "image_info"
+    description: <<END
+A 2-D float tensor of shape `[num_images, 5]` containing image information Height, Width, Scale.
+END
+  }
+  in_arg {
+    name: "anchors"
+    description: <<END
+A 2-D float tensor of shape `[num_anchors, 4]` describing the anchor boxes. Boxes are formatted in the form [y1, x1, y2, x2].
+END
+  }
+  in_arg {
+    name: "nms_threshold"
+    description: <<END
+A scalar float tensor for non-maximal-suppression threshold.
+END
+  }
+  in_arg {
+    name: "pre_nms_topn"
+    description: <<END
+A scalar int tensor for the number of top scoring boxes to be used as input.
+END
+  }
+  in_arg {
+    name: "min_size"
+    description: <<END
+A scalar float tensor. Any box that has a smaller size than min_size will be discarded.
+END
+  }
+  out_arg {
+    name: "rois"
+    description: <<END
+A 3-D float tensor of shape `[num_images,post_nms_topn,4]` representing the selected
+region of interest boxes. Sorted in descending order in scores.
+END
+  }
+  out_arg {
+    name: "roi_probabilities"
+    description: <<END
+A 2-D float tensor of shape `[num_images, post_nms_topn]` representing the score of the
+region of interest box in `rois` tensor at the same index.
+END
+  }
+  attr {
+    name: "post_nms_topn"
```

Comment:
This affects the output shape. I was told before that the attributes that change the output shape should be attributes not inputs. If I do it, this will be a data dependent op. Do you really want that?